src/main.rs

```diff
@@ -65,14 +68,26 @@ fn gumpi_async(opt: Opt, config: JobConfig) -> impl Future<Item = (), Error = fa
     // It's safe to call expect here - at this point opt.jobconfig
     // is guaranteed to be a valid filepath, which is checked by
     // JobConfig::from_file in main.rs
-    let sources_dir = opt
+    let jobconfig_dir = opt
```

Comment:
This looks good. Just a thought though on the best way of handling situations like this. Would you say that if we expect the input to be correct at this point, we should still run `expect` for better error message mainly, or an old-fashioned `unwrap` would be as good? I'm not looking for any changes here, I'm just curious to see what your thoughts on the topic are. :-)

Reply:
I'm still for using expect here. The fact that we expect the input to have been validated is an invariant that we want to keep. In the case we accidentally forget to keep this invariant, it would be nice to have a useful panic message when the program discovers it's broken.

Reply:
Yep, agreed!